File: seminar_2/task_3/shatrov.igor/Task3.java
import java.util.Scanner;

public class Task3 {
    public static void main(String[] args) {
        Scanner scn = new Scanner(System.in);

        int target = scn.nextInt();
        
        int left = -1, right = target + 1;

        while (left + 1 != right) {
            int mid = left + (right - left) / 2;
            
            if (mid * mid > target) {
                right = mid;
            } else {
                left = mid;
            }
        }

        System.out.println(left);
    }
}

File: seminar_2/task_3/gukasian.vladimir/task3.go
package main 

import "fmt" 

func binSearchSqrt(target int) int {
    var middle int

    right := target

    for left:=0; left <= right; {
        middle = (left + right) / 2

        if middle * middle > target {
            right = middle - 1
            continue
        }

        if middle * middle < target {
            left = middle + 1
            continue
        }

        return middle
    } 

    return right
}


func main() {
    fmt.Println(binSearchSqrt(17))
}
